lib/src/tinycolor.dart

```diff
@@ -125,38 +127,42 @@ class TinyColor {
   TinyColor desaturate([int amount = 10]) {
     final hsl = toHSLColor();
     final saturation = clamp01(hsl.saturation - amount / 100);
-    return TinyColor.fromHSL(hsl.withSaturation(saturation));
+    _color = hsl.withSaturation(saturation).toColor();
+    return this;
   }
 
   TinyColor saturate([int amount = 10]) {
     final hsl = toHSLColor();
     final saturation = clamp01(hsl.saturation + amount / 100);
-    return TinyColor.fromHSL(hsl.withSaturation(saturation));
+    _color = hsl.withSaturation(saturation).toColor();
+    return this;
   }
 
   TinyColor greyscale() => desaturate(100);
 
   TinyColor spin(double amount) {
     final hsl = toHSLColor();
     final hue = (hsl.hue + amount) % 360;
-    return TinyColor.fromHSL(hsl.withHue(hue < 0 ? 360 + hue : hue));
+    _color = hsl.withHue(hue < 0 ? 360 + hue : hue).toColor();
+    return this;
   }
 
   TinyColor mix(Color toColor, [int amount = 50]) {
     final p = amount / 100.0;
-    final color = Color.fromARGB(
+    _color = Color.fromARGB(
       ((toColor.alpha - _color.alpha) * p + _color.alpha).round(),
       ((toColor.red - _color.red) * p + _color.red).round(),
       ((toColor.green - _color.green) * p + _color.green).round(),
       ((toColor.blue - _color.blue) * p + _color.blue).round(),
     );
-    return TinyColor.fromColor(color);
+    return this;
```

Comment:
@WieFel Maybe I am not thinking straight yet. I am confused why the return value has changed as it shouldn’t.

Reply:
@calvintam236 does it only happen with `mix` or also with other functions?

Reply:
https://github.com/TinyCommunity/tinycolor2/runs/6042102828

`.shade()` failed due to using `.mix()`.

Reply:
@WieFel I am thinking could this part causing that?
See https://github.com/bgrins/TinyColor/blob/master/tinycolor.js#L38-L44.

Reply:
Yeah, looks indeed a bit strange.
I'm sorry for being so unresponsive currently... Very busy...